src/DotLiquid.Tests/Tags/IncludeTagTests.cs

```diff
@@ -139,6 +154,19 @@ public void TestIncludeTagWithLocalVariables()
             Assert.AreEqual("Locale: test123 ", Template.Parse("{% include 'locale_variables' echo1: 'test123' %}").Render());
         }
 
+        [Test]
+        public void TestIncludeTagWithStringVariable()
+        {
+            WithFileSystem(new DictionaryFileSystem(new Dictionary<string, string>
+            {
+                ["product"] = "Product: {{ product }}"
+            }), () =>
+            {
+                Assert.AreEqual("Product: foo", Template.Parse("{% include 'product' with 'foo' %}").Render());
+                Assert.AreEqual("Product: foo", Template.Parse("{% include 'product' for 'foo' %}").Render());

```

Comment:
I would suggest another test where string is not hard-coded and is a variable from hash.